toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -761,21 +762,28 @@ func updateInitramfsForEncrypt(installChroot *safechroot.Chroot) (err error) {
 	return
 }
 
-func updateFstab(installRoot string, installMap, mountPointToFsTypeMap, mountPointToMountArgsMap map[string]string) (err error) {
+func updateFstab(installRoot string, installMap, mountPointToFsTypeMap, mountPointToMountArgsMap map[string]string, hidepidEnable bool) (err error) {
 	ReportAction("Configuring fstab")
 
 	for mountPoint, devicePath := range installMap {
 		if mountPoint != "" && devicePath != NullDevice {
-			err = addEntryToFstab(installRoot, mountPoint, devicePath, mountPointToFsTypeMap[mountPoint], mountPointToMountArgsMap[mountPoint])
+			err = addEntryToFstab(installRoot, mountPoint, devicePath, mountPointToFsTypeMap[mountPoint], mountPointToMountArgsMap[mountPoint], false)
 			if err != nil {
 				return
 			}
 		}
 	}
+
+	if hidepidEnable {
+		err = addEntryToFstab(installRoot, "/proc", "proc", "proc", "rw,nosuid,nodev,noexec,relatime,hidepid=2", true)
+		if err != nil {
+			return
+		}
+	}
 	return
 }
 
-func addEntryToFstab(installRoot, mountPoint, devicePath, fsType, mountArgs string) (err error) {
+func addEntryToFstab(installRoot, mountPoint, devicePath, fsType, mountArgs string, doPseudoFsMount bool) (err error) {
```

Comment:
Please also mention the change related to `doPseudoFsMount` in the PR's summary or (even cleaner) move it to a separate PR.

Reply:
Keeping in this PR but did update the title

Reply:
Could you also extend the summary to explain the need for this change. Is it something the users should know about? Or to ask in a different way: should we mention the effect of this change in the documentation?

Reply:
At the moment, users should not see this change. I have plans to refactor this function to allow for users to configure this option in the future but that will be in a future PR :).  I can add some more detail to the summary